src/burnchains/affirmation.rs

```diff
@@ -0,0 +1,3315 @@
+        test_debug!("No block-commit has enough support to be an anchor block");
+        return Ok(None);
+    }
+
+    // find the ancestor that this tip confirms
+    let heaviest_ancestor_header = indexer
+        .read_burnchain_headers(ancestor_block, ancestor_block + 1)?
+        .first()
+        .expect(&format!(
+            "BUG: no block headers for height {}",
+            ancestor_block
+        ))
+        .to_owned();
+
+    let heaviest_ancestor_block =
+        BurnchainDB::get_burnchain_block(burnchain_tx.conn(), &heaviest_ancestor_header.block_hash)
+            .expect(&format!(
+                "BUG: no ancestor block {:?} ({})",
+                &heaviest_ancestor_header.block_hash, heaviest_ancestor_header.block_height
+            ));
+
+    // find the PoX anchor block-commit, if it exists at all
+    // (note that it may not -- a rich attacker can force F*w confirmations with lots of BTC on a
+    // commit that was never mined).
+    for block_op in heaviest_ancestor_block.ops.into_iter() {
+        if let BlockstackOperationType::LeaderBlockCommit(opdata) = block_op {
+            if opdata.block_height == ancestor_block && opdata.vtxindex == ancestor_vtxindex {
+                // found
+                debug!(
+                    "PoX anchor block-commit {},{},{} has {} burnt, {} confs",
+                    &opdata.txid, opdata.block_height, opdata.vtxindex, most_burnt, most_confs
+                );
+
+                let mut descendancy = Vec::with_capacity(prepare_ops.len());
+                for prepare_block_ops in prepare_ops.iter() {
+                    let mut block_descendancy = Vec::with_capacity(prepare_ops.len());
+                    for opdata in prepare_block_ops.iter() {
+                        if let Some((op_ancestor_height, op_ancestor_vtxindex, ..)) =
+                            ancestors.get(&(opdata.block_height, opdata.vtxindex))
+                        {
+                            if *op_ancestor_height == ancestor_block
+                                && *op_ancestor_vtxindex == ancestor_vtxindex
+                            {
+                                test_debug!("Block-commit {},{} descends from likely PoX anchor block {},{}", opdata.block_height, opdata.vtxindex, op_ancestor_height, op_ancestor_vtxindex);
+                                block_descendancy.push(true);
+                            } else {
+                                test_debug!("Block-commit {},{} does NOT descend from likely PoX anchor block {},{}", opdata.block_height, opdata.vtxindex, ancestor_block, ancestor_vtxindex);
+                                block_descendancy.push(false);
+                            }
+                        } else {
+                            test_debug!("Block-commit {},{} does NOT descend from likely PoX anchor block {},{}", opdata.block_height, opdata.vtxindex, ancestor_block, ancestor_vtxindex);
```

Comment:
These seem like they would be usefully `debug` instead of `test_debug`

Reply:
Done